tests/Api/Response/Parser/FiltersParserTest.php

```diff
@@ -0,0 +1,299 @@
+
+use Findologic\Api\Response\Parser\FiltersParser;
+use Findologic\Api\Response\Response;
+use Findologic\Api\Services\ColorImage;
+use Plenty\Modules\Plugin\Libs\Contracts\LibraryCallContract;
+use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\MockObject\MockObject;
+
+/**
+ * Class FiltersParserTest
+ * @package Findologic\Tests\Api\Response\Parser
+ */
+class FiltersParserTest extends TestCase
+{
+    /**
+     * @var ColorImage|MockObject
+     */
+    protected $colorImage;
+
+    /**
+     * @var LibraryCallContract|MockObject
+     */
+    protected $libraryCallContract;
+
+    public function setUp()
+    {
+        $this->libraryCallContract = $this->getMockBuilder(LibraryCallContract::class)
+            ->disableOriginalConstructor()
+            ->setMethods([])
+            ->getMock();
+        $this->colorImage = $this->getMockBuilder(ColorImage::class)
+            ->disableOriginalConstructor()
+            ->setMethods([])
+            ->getMock();
+    }
+
+    public function testParse()
+    {
+        $responseMock = $this->getMockBuilder(Response::class)->disableOriginalConstructor()->setMethods(null)->getMock();
+        /** @var FiltersParser|MockObject $filtersParserMock */
+        $filtersParserMock = $this->getFiltersParserMock(['createResponseObject']);
+        $filtersParserMock->expects($this->any())->method('createResponseObject')->willReturn($responseMock);
+
+        $results = $filtersParserMock->parse($this->getNoFiltersResponse());
+        $this->assertEmpty($results);
+
+        $response = $this->getResponse();
+        $results = $filtersParserMock->parse($response);
+
+        $this->assertNotEmpty($results);
+
+        foreach ($results as $index => $result) {
+            $this->assertEquals($result['id'], $response->filters->filter[$index]->name);
+            $this->assertEquals($result['name'], $response->filters->filter[$index]->display);
+            $this->assertEquals($result['select'], $response->filters->filter[$index]->select);
+            $this->assertEquals($result['type'], $response->filters->filter[$index]->type);
+            $this->assertEquals(count($result['values']), count($response->filters->filter[$index]->items->item));
+        }
+    }
+
+    /**
+     * @param array|null $methods
+     * @return FiltersParser|MockObject
+     */
+    protected function getFiltersParserMock($methods = null)
+    {
+        $filtersParserMock = $this->getMockBuilder(FiltersParser::class)
+            ->setConstructorArgs([
+                'libraryCallContract' => $this->libraryCallContract,
+                'colorImageService' => $this->colorImage
+            ])
+            ->setMethods($methods);
+
+        return $filtersParserMock->getMock();
+    }
+
+    /**
+     * @return \SimpleXMLElement
+     */
+    protected function getNoFiltersResponse()
+    {
+        return simplexml_load_string(preg_replace('/&(?!#?[a-z0-9]+;)/', '&amp;','<?xml version="1.0"?>
```

Comment:
Why is the `preg_replace` necessary?

Reply:
I have removed _preg_replace_ from _getNoFiltersResponse()_, since there are no ampersands. There are ampersands in _getResponse()_, and it is not parsed by _simplexml_load_string_, so needed to replace it with &amp; .

Reply:
I have added preg_replace to replace ampersand symbol with html friendly ampersand. But now I have removed preg_replace and changed the values itself in xml feed, since it is testing data, so I guess it will be clearer.